src/routes/blogs.rs

```diff
@@ -149,6 +151,102 @@ pub fn delete(name: String, rockets: PlumeRocket) -> Result<Redirect, Ructe>{
     }
 }
 
+#[derive(FromForm, Validate)]
+pub struct EditForm {
+    #[validate(custom(function = "valid_slug", message = "Invalid name"))]
+    pub title: String,
+    pub summary: String,
+    pub icon: Option<i32>,
+    pub banner: Option<i32>,
+}
+
+#[get("/~/<name>/edit")]
+pub fn edit(conn: DbConn, name: String, user: Option<User>, intl: I18n) -> Result<Ructe, ErrorPage> {
+    let blog = Blog::find_by_fqn(&*conn, &name)?;
+    if user.clone().and_then(|u| u.is_author_in(&*conn, &blog).ok()).unwrap_or(false) {
+        let user = user.expect("blogs::edit: User was None while it shouldn't");
+        let medias = Media::for_user(&*conn, user.id).expect("Couldn't list media");
+        Ok(render!(blogs::edit(
+            &(&*conn, &intl.catalog, Some(user)),
+            &blog,
+            medias,
+            &EditForm {
+                title: blog.title.clone(),
+                summary: blog.summary.clone(),
+                icon: blog.icon_id,
+                banner: blog.banner_id,
+            },
+            ValidationErrors::default()
+        )))
+    } else {
+        // TODO actually return 403 error code
+        Ok(render!(errors::not_authorized(
+            &(&*conn, &intl.catalog, user),
+            i18n!(intl.catalog, "You are not allowed to edit this blog.")
+        )))
+    }
+}
+
+/// Returns true if the media is owned by `user` and is a picture
+fn check_media(conn: DbConn, id: i32, user: &User) -> bool {
```

Comment:
Does this function actually do what it says it does?

Reply:
Looks like I pushed too fast… :sweat_smile: